### ISSUE TITLE
Installing Arch Linux on a Microsoft Surface 3

### ISSUE THREAD

**Issue by seanlane** [seanlane/seanlane-comments#6]
Comments for my blog post on running Arch Linux on a Microsoft Surface 3: http://sean.lane.sh/blog/2018/Arch_linux-on_microsoft_surface_3

**Comment by rbn920** [comment]
I was wondering how this is working out (or if it is). Are you still on that old kernel?

**Comment by seanlane** [comment]
It's a little rough, but it's still working. Every once in a while, something becomes unresponsive, and a reboot seems to help. I recently was notified of this comment indicating that kernel 5.1.15 may also work, but I haven't tested it myself: 

https://github.com/jakeday/linux-surface/issues/452#issuecomment-576092252

**Comment by eduncan911** [comment]
His repo has long been abandoned.

Several people took his patches along with qzen's (the one that continued development on this kernel) and created a new GitHub org/team with several people, in the event someone becomes unavailable to continue.

https://github.com/linux-surface/linux-surface

Not only that, but they really went the extra *extra* mile by:

* Building a full CICD system that takes the latest kernel when released, patches, abs publishes automatically.
* Full package repo support for most base (Linux) distros: Arch, Debian, Fedora
* Most up to date kernel.  5.6 was released in mid-March, they had the first builds published 12 days later on `linux-surface`.

I'm running 5.6.5 on Pop!_OS (19.10), even though it shipped with 5.3.

I myself have attempted many times to setup our old Surface 3 LTE 4GB 128GB (not free, but a $250 closeout) with Linux, often failing at keeping the Wireless working between sleep and wake.  Arch, Elementary, Ubuntu, Pop_OS, etc.

It wasn't until today when revisiting the `linux-surface` status that I found that not only have they identified the battery and sound issues (requires a BIOS update, which they have instructions for on that site to do - from within Linux and no Windows!), but also includes the latest kernel for the Battery fix as well!

https://github.com/linux-surface/linux-surface/wiki/Surface-3

I am having a few issues though...

* Random gnome freezes/crashes/restarts, sometimes requiring a full restart of the OS.
* Rotation was working, and still is with the login.  However, it's stopped within Gnome with Pop!_OS out of the box.  Most likely an update broke the rotation.
* Keyboard /trackpad doesn't seem to come back after folding back, and then bringing to front to reuse.  Logs show an SPI bus communication issue. 

Anyhoot, it's very usable now.

---

So a question: what DE are you using with ElementaryOS?  Did you stick to Xfce?

If so, how did you setup all of the touch controls on log, screensaver, suspend/sleep, and general usage?

So, under Gnome we can use 150% scaling which seems to be the sweet spot for my kiddo and me, though 175% seems easier on my eyes.  200% is too large.

Gnome supports scaling of almost everything.  However, under Xfce and Arch I found myself spending all nighters tracking down and resolving each app's scaling issues, one by one.  I don't have that energy to loose any longer.

Hence, why doing with Pop!_OS.  However, general is slow as sh*t.

**Comment by seanlane** [comment]
Wow, this is a great update, thanks for posting! I did my original write up about a 18 months before the jakeday repo went dark, so I'm out of date as well. 

I want to say that the Elementary OS install is using Xfce, since I thought the simpler DE would perform better on the Surface 3. I'm not entirely sure though, it's been a bit. It was fairly plug and play, touch did work roughly out of the box, but we mostly stuck to using the trackpad.

I'm wrapping up a thesis at the moment, but when I have the chance I'll circle back to this and update with my own experiences.

**Comment by rbn920** [comment]
Yes, I had noticed that as well. After following a trail of archived github repos, reddit posts, defuncked gitlab pages, etc, I finally found the repo mentioned above. I'm all set up running Fedora and it seems to be working decently.

**Comment by eduncan911** [comment]
Yeah, I've been trying to use that newer linux-surface kernel with PoP!_OS 19.04 and 19.10 (Ubuntu under the hood); but after many wipes and bare-bone installs, Gnome just randomly freezes after 30m or more of usage usually via touchscreen.  

Finally just installed Ubuntu 20.04, with that latest linux-surface kernel - zero issues and long battery life.  I tried the updated 4.19 kernel, but battery wasn't working and pen didn't register property.  So just running the latest and the pen and eraser all works.

**Comment by misterchalm22** [comment]
> Yeah, I've been trying to use that newer linux-surface kernel with PoP!_OS 19.04 and 19.10 (Ubuntu under the hood); but after many wipes and bare-bone installs, Gnome just randomly freezes after 30m or more of usage usually via touchscreen.
> 
> Finally just installed Ubuntu 20.04, with that latest linux-surface kernel - zero issues and long battery life. I tried the updated 4.19 kernel, but battery wasn't working and pen didn't register property. So just running the latest and the pen and eraser all works.

As a Ubuntu noob, how would I install 20.04 with that kernel? I assume its not included in the installation package by default...

**Comment by eduncan911** [comment]
> > Yeah, I've been trying to use that newer linux-surface kernel with PoP!_OS 19.04 and 19.10 (Ubuntu under the hood); but after many wipes and bare-bone installs, Gnome just randomly freezes after 30m or more of usage usually via touchscreen.
> > Finally just installed Ubuntu 20.04, with that latest linux-surface kernel - zero issues and long battery life. I tried the updated 4.19 kernel, but battery wasn't working and pen didn't register property. So just running the latest and the pen and eraser all works.
> 
> As a Ubuntu noob, how would I install 20.04 with that kernel? I assume its not included in the installation package by default...

First of all, go to the right repo.  This is the wrong one and no longer supported.  Go here instead:

https://github.com/linux-surface/linux-surface

After following their instructions to setup a PPA for your Ubuntu instance (it's all there), then drill down to the Surface 3 specifics for final tweaks.

https://github.com/linux-surface/linux-surface/wiki/Surface-3

**Comment by misterchalm22** [comment]
Thanks!! This S3 speed is really getting me down. Hopefully Linux can fox it up so it's usable again.

**Comment by eduncan911** [comment]
@misterchalm22 IMO, it wasn't much different in speed.  Sure, the UX is snappy (and more so under Windows 10 than Ubuntu 20.04 with hardware acceleration).  but when it comes to anything, like opening Blender, or just trying to draw with the stylus in Blender, the S3 is just so painfully slow beyond usability (for 3D modelling).

I basically switched our kid to a Core i7 tablet I had left over, and took the S3 back as my lightweight reading and movie watching device - running Windows 10, as there's just those ever slight quirks still remaining with Linux.
